### PR TITLE
Proper strings output args handling in opcode 0ad4

### PR DESCRIPTION
**Pull request by MiranDMC**
Added standard CLEO5 opcode param types validation.
Fixed possible buffer overruns when target types are variables strings.

### CODE REVIEW

File: source/CCustomOpcodeSystem.cpp [hunk 1]
```diff
@@ -1477,39 +1477,67 @@ namespace CLEO
...
 		{
 			auto paramType = thread->PeekDataType();
-			if (paramType != DT_END)
+
+			if (paramType == DT_END)
 			{
-				ExParams[i] = GetScriptParamPointer(thread);
-				cExParams++;
+				outputParams[i] = nullptr;
+				continue;
```
Comment (x87) [inline source/CCustomOpcodeSystem.cpp:1499]: break?

Comment (x87) [reply source/CCustomOpcodeSystem.cpp:1499]: nvm

File: source/CCustomOpcodeSystem.cpp [hunk 2]
```diff
@@ -1477,39 +1477,67 @@ namespace CLEO
...
+						 outputParams[12], outputParams[13], outputParams[14], outputParams[15], outputParams[16], outputParams[17],
+						 outputParams[18], outputParams[19], outputParams[20], outputParams[21], outputParams[22], outputParams[23],
+						 outputParams[24], outputParams[25], outputParams[26], outputParams[27], outputParams[28], outputParams[29],
+						 outputParams[30], outputParams[31], outputParams[32], outputParams[33], outputParams[34]);
+
+		// transfer string params to target variables
+		for (auto& p : stringParams)
+		{
+			if (p.used) WriteStringParam(p.target, p.str.c_str());
+		}
```
Comment (x87) [inline source/CCustomOpcodeSystem.cpp:1538]: tests maybe?

Comment (MiranDMC) [reply source/CCustomOpcodeSystem.cpp:1538]: yeah, will do

**Comment by MiranDMC** [comment]
Implemented in https://github.com/cleolibrary/CLEO5/pull/104

**Pull request closed without merging**